warehouse/macaroons/caveats.py

```diff
@@ -42,33 +48,75 @@ def verify_projects(self, projects):
             )
 
         project = self.verifier.context
-        if project.normalized_name in projects:
-            return True
 
+        for user_proj in projects:
```

Comment:
I think we need to make this loop more flexible to handle API tokens that are already out there in the world: we should check if each element is a `str` -- if it is, then treat is as the normalized name. If it's a `dict`, then perform the `project-name` lookup.